Multiple #version in GLSL shaders when BUFFER_RO or BUFFER_RW is used

I'm trying to use shaders from here: https://github.com/pezcode/Cluster

When i'm including [clusters.sh](https://github.com/pezcode/Cluster/blob/94c0c0aa37ab3f14071aca16f2dbf171514612c6/src/Renderer/Shaders/clusters.sh) into a fragment shader to visualize rendering in debug, it pulls `BUFFER_RO` or `BUFFER_RW` which are `__BUFFER_XX` macro internally (in `bgfx_compute.sh`), but with different access. As far as i can tell, `shaderc` sees that there is `layout(std430` string in the shader and adds `#version 430` and some other stuff like this:

```c
#version 430
#define texture2DLod      textureLod
#define texture2DGrad     textureGrad
#define texture2DProjLod  textureProjLod
#define texture2DProjGrad textureProjGrad
...
```

Then, it seems that `gl::ShaderGL::create()` checks if `BGFX_CONFIG_RENDERER_OPENGL` is set to >= 31 (i have it set to 43) and if it so then adds `#version 140` on top of the shader code, so final shader code becomes like this:

```c
#version 140
...
#version 430
...
```

And i'm getting `error C0204: version directive must be first statement and may not be repeated` (on NVidia).

This is also seems to be the case for vertex shaders, although effect might be different, there are some checks for vertex/fragment shader in OpenGL rendered, i think different text might be added to the beginning of shader code depending on shader type.

I've made some ugly hacks to bypass this based on this line:

```c++
const bool versionSet = (0 == bx::strCmp(code, "#version", 8));
```

Generally if version is already set, then no other code will be added to shader. After this fragment shader compiled and seemingly executed. But maybe it needs another branch for `BGFX_CONFIG_RENDERER_OPENGL >= 43`.

Am i doing something wrong? Am i supposed to use `BUFFER_RO` or `BUFFER_RW` in fragment shader?

--- a/src/renderer_gl.cpp
+++ b/src/renderer_gl.cpp
@@ -5375,7 +5375,8 @@ namespace bgfx { namespace gl
 
 		if (0 != m_id)
 		{
-			if (GL_COMPUTE_SHADER != m_type)
+			if (GL_COMPUTE_SHADER != m_type
+			&&  0 != bx::strCmp(code, "#version 430", 12) ) // #2000
 			{
 				int32_t tempLen = code.getLength() + (4<<10);
 				char* temp = (char*)alloca(tempLen);
@@ -5873,7 +5874,7 @@ namespace bgfx { namespace gl
 
 				code.set(temp);
 			}
-			else // GL_COMPUTE_SHADER
+			else if (GL_COMPUTE_SHADER == m_type)
 			{
 				int32_t codeLen = (int32_t)bx::strLen(code);
 				int32_t tempLen = codeLen + (4<<10);

